parcels/codegenerator.py

```diff
@@ -164,6 +164,8 @@ def __init__(self, grid, ptype):
         self.grid = grid
         self.ptype = ptype
         self.field_args = OrderedDict()
+        # Hack alert: JIT requires U field to update grid indexes
+        self.field_args['U'] = grid.U
```

Comment:
Yes, it's totally hacky, but the underlying JIT code to update `p->xi` and `p->yi` always uses `U`, which is already a quite hacky. I'll clean that up soon hopefully and do a more generalised field -> index mapping.